### PR TITLE
chore: 🤖 update ember-cli-mirage

### PR DESCRIPTION
**Pull request by cameronperera**
# Description
This PR updates `ember-cli-mirage` to v3.0.3. This is a minor change that will help with updating packages for this [dependabot alert](https://github.com/hashicorp/boundary-ui/security/dependabot/140). 

## Screenshots (if appropriate)

I saw this warning while running test. I removed the entry for fireworm from yarn.lock [here](https://github.com/hashicorp/boundary-ui/pull/2458/files#diff-51e4f558fae534656963876761c95b83b6ef5da5103c4adef6768219ed76c2deR13025-R13027) and ran `yarn install` again. This cleared up the warning.
Screenshot 2024-08-22 at 4 55 40 PM

## How to Test

Please navigate around the Admin UI and desktop client Vercel builds. I have done local testing and have not see any issues and this [build](https://github.com/hashicorp/boundary-ui-releases/actions/runs/10529040740) was successful.

## Checklist

- [ ] I have added before and after screenshots for UI changes
- [ ] I have added JSON response output for API changes
- [ ] I have added steps to reproduce and test for bug fixes in the description
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: addons/api/package.json [hunk 1]
```diff
@@ -49,6 +50,7 @@
     "ember-cli": "^4.12.1",
     "ember-cli-dependency-checker": "^3.3.1",
     "ember-cli-inject-live-reload": "^2.1.0",
+    "ember-cli-mirage": "3.0.3",
```
Comment (ZedLi) [inline addons/api/package.json:53]: We have this now both in `devDependencies` and `dependencies`

**Review by ZedLi — CHANGES_REQUESTED** [review]

File: addons/api/package.json (hunk 1 above)
Comment (cameronperera) [reply addons/api/package.json:53]: Apologies. I will fix that now.

File: addons/api/package.json [hunk 2]
```diff
@@ -41,6 +41,7 @@
     "@ember/string": "^3.0.1",
     "@ember/test-helpers": "^2.9.3",
     "@embroider/test-setup": "^2.1.1",
+    "@faker-js/faker": "^8.0.2",
```
Comment (calcaide) [inline addons/api/package.json:44]: What is the reason for adding faker here, it was not here before, right?

Comment (calcaide) [reply addons/api/package.json:44]: Oh I see before it was on line 67 and now we move it to 44?

**Comment by calcaide** [comment]
@cameronperera 
I do see the build you link within the description has an error when generating release metadata. I am pretty sure it is not related, but I will feel more confident to approve this PR if we can do a run with all green (or just re-run the failed step there).

For context:
The release metadata geneartion step is prettys straight forward, B-U-T the piece of software we use to do so is very picky (as it should) with namings, arch's and platforms. Sometimes, this job will fail because we changed something that affected the naming generation of a zip or a binary, which usually means we broke something else, that's why my comment above. I am pretty sure in this case is a hiccup, but I would like if you can re-run.

**Review by calcaide — DISMISSED** [review]
Left a couple of commetns

File: addons/api/package.json (hunk 2 above)
Comment (cameronperera) [reply addons/api/package.json:44]: yeah I am not exactly sure why it moved. It just did.

**Comment by cameronperera** [comment]
> @cameronperera I do see the build you link within the description has an error when generating release metadata. I am pretty sure it is not related, but I will feel more confident to approve this PR if we can do a run with all green (or just re-run the failed step there).
> 
> For context: The release metadata geneartion step is prettys straight forward, B-U-T the piece of software we use to do so is very picky (as it should) with namings, arch's and platforms. Sometimes, this job will fail because we changed something that affected the naming generation of a zip or a binary, which usually means we broke something else, that's why my comment above. I am pretty sure in this case is a hiccup, but I would like if you can re-run.

I fixed the naming of my test release branch and that fixed the issue. I updated the description with the new build and it was successful all the way through!

File: addons/api/package.json (hunk 1 above)
Comment (ZedLi) [reply addons/api/package.json:54]: I'm confused, why do we now have one pinned to `3.0.3` in `devDependencies` and `^3.0.3` in `dependencies`?

Comment (cameronperera) [reply addons/api/package.json:54]: I did some git stashing and I missed the devDep one. Let me try this again. Apologies, I thought I had it all fixed.

Comment (cameronperera) [reply addons/api/package.json]: Fixed and test build ran successfully and local tests working.

**Review by lisbet-alvarez — APPROVED** [review]
Thanks for updating this!

**Review by ZedLi — APPROVED** [review]

File: addons/api/package.json (hunk 2 above)
Comment (ZedLi) [reply addons/api/package.json:45]: It's because it's supposed to be alphabetical ordering so it re-ordered it

Comment (cameronperera) [reply addons/api/package.json:45]: Oh yeah. I would have expected to see that more often though so I didn't think to check for the order.

**Pull request merged**